californium-core/src/main/java/org/eclipse/californium/core/coap/Request.java

```diff
@@ -1128,6 +1131,36 @@ public void setSendError(Throwable sendError) {
 		}
 	}
 
+	/**
+	 * Return the error which happened on response handling. 
+	 * 
+	 * @return a {@link Throwable} if an error happened or {@code null} is there was no error.
+	 */
+	public Throwable getOnResponseError() {
+		return responseHandlingError;
+	}
+
+	/**
+	 * Mark this request as failing to handle the response.
+	 * 
+	 * @param cause the cause of the failure.
+	 */
+	public void setOnResponseError(Throwable cause) {
+		this.responseHandlingError = cause;
+		if (responseHandlingError != null) {
+			for (MessageObserver handler : getMessageObservers()) {
+				if (handler instanceof MessageObserver2) {
```

Comment:
Maybe the if could be moved out of the loop.

Reply:
If not sure to get this :thinking:

Reply:
Ah, I'm wrong :-)